src/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java

```diff
@@ -463,44 +463,75 @@ public void publishAssociated(Contentlet contentlet, boolean isNew) throws DotSe
     }
 
     @Override
-    public void publishAssociated(Contentlet contentlet, boolean isNew, boolean isNewVersion) throws DotSecurityException, DotDataException,
-    DotContentletStateException, DotStateException {
+    public void publishAssociated(Contentlet contentlet, boolean isNew, boolean isNewVersion) throws
+        DotSecurityException, DotDataException, DotStateException {
 
         if (!contentlet.isWorking())
             throw new DotContentletStateException("Only the working version can be published");
 
         // writes the contentlet object to a file
         indexAPI.addContentToIndex(contentlet, true, true);
 
-        User user = APILocator.getUserAPI().getSystemUser();
-
         // DOTCMS - 4393
         // Publishes the files associated with the Contentlet
         List<Field> fields = FieldsCache.getFieldsByStructureInode(contentlet.getStructureInode());
         Language defaultLang=APILocator.getLanguageAPI().getDefaultLanguage();
+        User systemUser = APILocator.getUserAPI().getSystemUser();
+
         for (Field field : fields) {
-            if (field.getFieldType().equals(Field.FieldType.IMAGE.toString())
-                    || field.getFieldType().equals(Field.FieldType.FILE.toString())) {
+            if (field.getFieldType().equals(Field.FieldType.IMAGE.toString()) ||
+                field.getFieldType().equals(Field.FieldType.FILE.toString())) {
 
+                // I know! You already saw the nested try/catch blocks below,
+                // please don't shoot the messenger, let me explain.
+                // NOTE: Keep in mind that at this moment the FILE ASSET could be in the same language or
+                // default lang (DEFAULT_FILE_TO_DEFAULT_LANGUAGE=true)
                 try {
-                    String value = "";
-                    if(UtilMethods.isSet(getFieldValue(contentlet, field))){
-                        value = getFieldValue(contentlet, field).toString();
-                    }
-                    Identifier id = APILocator.getIdentifierAPI().find(value);
-                    if (InodeUtils.isSet(id.getInode()) && id.getAssetType().equals("contentlet")) {
+                    // We need to get the Identifier from the field. (Image or File)
+                    String fieldValue = UtilMethods.isSet(getFieldValue(contentlet, field)) ?
+                        getFieldValue(contentlet, field).toString() : "";
```

Comment:
you can use StringUtils.BLANK either StringPools instead of creating a new empty string